src/ui/common/src/components/integrations/dialogs/dialog.tsx

```diff
@@ -38,149 +40,58 @@ import { RedshiftDialog } from './redshiftDialog';
+    editMode ? integrationToEdit.name : ''
   );
 
-  const confirmConnect = () => {
-    setIsConnecting(true);
-    setErrMsg(null);
-    addTable(user, integrationId, config)
-      .then(() => {
-        setShowSuccessToast(true);
-        const successMessage =
-          'Successfully uploaded CSV file to the demo database!';
-        setSuccessMessage(successMessage);
-        onConnect();
-        setIsConnecting(false);
-      })
-      .catch((err) => {
-        const errorMessage = 'Unable to upload CSV file to the demo database: ';
-        setErrMsg(errorMessage + err.message);
-        setIsConnecting(false);
-      });
-  };
-
-  return (
-    <Dialog open={true} onClose={onCloseDialog} fullWidth maxWidth="lg">
-      <DialogTitle>{dialogHeader}</DialogTitle>
-      <DialogContent>
-        {serviceDialog}
-        {errMsg && <Alert severity="error">{errMsg}</Alert>}
-        <Snackbar
-          anchorOrigin={{ vertical: 'top', horizontal: 'center' }}
-          open={showSuccessToast}
-          onClose={handleSuccessToastClose}
-          key={'integrations-dialog-success-snackbar'}
-          autoHideDuration={6000}
-        >
-          <Alert
-            onClose={handleSuccessToastClose}
-            severity="success"
-            sx={{ width: '100%' }}
-          >
-            {successMessage}
-          </Alert>
-        </Snackbar>
-      </DialogContent>
-      <DialogActions>
-        <Button autoFocus onClick={onCloseDialog}>
-          Cancel
-        </Button>
-        <LoadingButton
-          autoFocus
-          onClick={confirmConnect}
-          loading={isConnecting}
-          disabled={disableConnect}
-        >
-          Confirm
-        </LoadingButton>
-      </DialogActions>
-    </Dialog>
+  const connectNewStatus = useSelector(
+    (state: RootState) => state.integrationReducer.connectNewStatus
   );
-};
-
-type IntegrationDialogProps = {
-  user: UserProfile;
-  service: Service;
-  onCloseDialog: () => void;
-};
-
-export const IntegrationDialog: React.FC<IntegrationDialogProps> = ({
-  user,
-  service,
-  onCloseDialog,
-}) => {
-  const dispatch: AppDispatch = useDispatch();
-  const navigate = useNavigate();
-  const [config, setConfig] = useState<IntegrationConfig>({});
-  const [disableConnect, setDisableConnect] = useState(true);
-  const [successMessage, setSuccessMessage] = useState('');
-  const [showSuccessToast, setShowSuccessToast] = useState(false);
 
-  const namePrefix = formatService(service) + '/';
-  const [name, setName] = useState(namePrefix);
+  const editStatus = useSelector(
+    (state: RootState) => state.integrationReducer.editStatus
+  );
 
-  const handleSuccessToastClose = () => {
-    setShowSuccessToast(false);
-  };
+  const connectStatus = editMode ? editStatus : connectNewStatus;
+  const disableConnect =
+    !editMode &&
+    (!isConfigComplete(config) || name === '' || name === aqueductDemoName);
+  const setConfigField = (field: string, value: string) =>
```

Comment:
this is basically `config[field] = value` in the language of state update

Reply:
Yep, nice use of the spread operator here too!